### PR TITLE
Fix format_bytes() units

### PR DESCRIPTION
**Pull request by martinpitt**
See individual commits for details.

----

## cockpit.js API: Fix format_bytes() units

The [`cockpit.format_bytes()`](https://cockpit-project.org/guide/latest/cockpit-util.html#cockpit-format-bytes) function now writes the unit for kilobytes as SI conformant "kB" instead of "KB" when formatting decimal (base-10) sizes. It also uses the unit "B" (bytes) for numbers less than 1000 instead of no unit at all.

Note that many projects, in particular the ones based on current [starter-kit](https://github.com/cockpit-project/starter-kit), bundle `cockpit.js` and usually also pin down a specific git commit from cockpit.git. So these projects won't immediately pick up the change, but should update the cockpit library import and possibly adjust their usage of `format_bytes()`.

### CODE REVIEW

File: pkg/kdump/kdump-view.jsx [hunk 1]
```diff
@@ -483,15 +483,15 @@ ${enableCrashKernel}
...
-            // error while reading
-        } else {
-            // assume we have a proper value
+        } else if (Number.isInteger(this.props.reservedMemory)) {
             // TODO: hint at using debug_mem_level to identify actual memory required?
-            reservedMemory = <span>{this.props.reservedMemory}</span>;
+            reservedMemory = <span>{cockpit.format_bytes(this.props.reservedMemory, { base2: true })}</span>;
+        } else {
+            // error while reading
+            reservedMemory = null;
```
Comment (cockpituous) [inline pkg/kdump/kdump-view.jsx:494]: This added line is not executed by any test.

File: pkg/kdump/kdump.js [hunk 2]
```diff
@@ -99,11 +99,11 @@ const initStore = function(rootElement) {
                 // https://access.redhat.com/solutions/59432 states limit to be 896MiB and the auto at 768MiB max
                 // default unit is MiB
                     if (value >= 1024 * 1024)
-                        dataStore.kdumpMemory = cockpit.format_bytes(value, 1024);
+                        dataStore.kdumpMemory = value;
                     else if (value >= 1024)
-                        dataStore.kdumpMemory = cockpit.format_bytes(value * 1024, 1024);
+                        dataStore.kdumpMemory = value * 1024;
```
Comment (cockpituous) [inline pkg/kdump/kdump.js:104]: This added line is not executed by any test.

File: pkg/kdump/kdump.js [hunk 3]
```diff
@@ -99,11 +99,11 @@ const initStore = function(rootElement) {
...
                     else
-                        dataStore.kdumpMemory = cockpit.format_bytes(value * 1024 * 1024, 1024);
+                        dataStore.kdumpMemory = value * 1024 * 1024;
```
Comment (jelly) [inline pkg/kdump/kdump.js:106]:
```
pkg/kdump/kdump.js:            reservedMemory: dataStore.kdumpMemory,
```
Ah that's how it works, funny level of indirection.

**Review by jelly — APPROVED** [review]
Nice cleanups and TIL https://en.wikipedia.org/wiki/Metric_prefix

File: pkg/kdump/kdump.js (hunk 3 above)
Comment (jelly) [reply pkg/kdump/kdump.js:106]: Also hmm:
```
pkg/kdump/kdump.js:            .catch(() => { dataStore.kdumpMemory = "error" });
```

File: pkg/base1/test-format.ts [hunk 4]
```diff
@@ -84,11 +84,11 @@ QUnit.test("format_number", function (assert) {
 
 QUnit.test("format_bytes", function (assert) {
     const checks = [
-        [5, 1000, "5"],
-        [5, 1024, "5"],
-        [999, 1000, "999"],
-        [999, 1024, "999"],
-        [1023, 1024, "1023"],
+        [5, 1000, "5 B"],
```
Comment (allisonkarlitskaya) [inline pkg/base1/test-format.ts:87]: I might have liked to have seen a test here that `0` → `0 B`, but it's covered by integration tests...

**Review by allisonkarlitskaya — APPROVED** [review]
Looks great, thanks!

Can you open issues in cpckpit-files and cockpit-podman when this lands?

File: pkg/base1/test-format.ts (hunk 4 above)
Comment (martinpitt) [reply pkg/base1/test-format.ts:87]: Ah, that's a good idea! I plan to do a follow-up anyway to clean up our deprecated ways of calling format_bytes() throughout -- I'll do that there!

Comment (martinpitt) [reply pkg/base1/test-format.ts:87]: Added in PR #20361

**Comment by martinpitt** [comment]
@allisonkarlitskaya :

> Can you open issues in cpckpit-files and cockpit-podman when this lands?

Fixed in https://github.com/cockpit-project/cockpit-files/pull/396

podman is more involved, see https://github.com/cockpit-project/cockpit-podman/pull/1673

**Pull request merged**